dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/osgi/request/IRequest60Impl.java

```diff
@@ -12,30 +12,66 @@
 
 import com.ibm.websphere.ras.Tr;
 import com.ibm.websphere.ras.TraceComponent;
-import com.ibm.websphere.servlet40.IRequest40;
 import com.ibm.ws.webcontainer.osgi.osgi.WebContainerConstants;
 import com.ibm.ws.webcontainer40.osgi.request.IRequest40Impl;
 import com.ibm.wsspi.http.HttpInboundConnection;
+import com.ibm.wsspi.http.ee7.HttpInboundConnectionExtended;
 
 import io.openliberty.websphere.servlet60.IRequest60;
 
-
-/**
- *
- */
 public class IRequest60Impl extends IRequest40Impl implements IRequest60 {
-
     private static final TraceComponent tc = Tr.register(IRequest60Impl.class, WebContainerConstants.TR_GROUP, WebContainerConstants.NLS_PROPS);
 
-    /**
-     * @param connection
-     */
     public IRequest60Impl(HttpInboundConnection connection) {
         super(connection);
 
         if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-            Tr.debug(tc, "constructor", " inboundConnection [" + connection + "]");
+            Tr.debug(tc, "constructor , inboundConnection [" + connection + "]");
+        }
+
+    }
+
+    @Override
+    public int getH2StreamId() {
+        int id = -1;
+        HttpInboundConnectionExtended ice;
+
+        if ((ice = getICE()) != null) {
+            id = ice.getH2StreamId();
+        }
+
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "getH2StreamId , returns stream id [" + id + "]");
+        }
+
+        return id;
+    }
+
+    @Override
+    public int getConnectionId() {
+        int id = -1;
+        HttpInboundConnectionExtended ice;
+
+        if ((ice = getICE()) != null) {
```

Comment:
In the case that this is null, however unlikely it may be, are consumers expected to know how to interpret a -1 value? Should the logging in this method call particular attention to when it is -1 or does it printing -1 for the id suffice for debugging?

Reply:
API is defined here :https://jakarta.ee/specifications/servlet/6.0/apidocs/jakarta.servlet/jakarta/servlet/servletconnection#getConnectionId()

Reply:
It is very unlikely to get here without a Http Inbound connection.
-1 can be used by the consumer of this interface for further interpretation.  So far, it is only consumed by our internal SRTServletRequest.